plugin/src/lib.rs

```diff
@@ -724,8 +724,8 @@ fn fold_first_block(block: P<Block>, m: &mut MutatorPlugin) -> P<Block> {
                         .unwrap(),
                 );
             }
-            for (ref key, ref values) in (&interchangeables).iter() {
-                for value in (&values).iter() {
+            for (ref key, ref values) in interchangeables {
+                for value in values.iter() {
```

Comment:
@llogiq - does the for loop that I implemented here looks correct?

Reply:
A `&` before `interchangeables` will help here, as we don't need to consume it. This will use `&Vec`s `IntoIterator` implemenation, so you don't need to call `.iter()` (which is another way to do the same thing)

Reply:
@llogiq - I tried that too `for (ref key, ref values) in &interchangeables {` but got compilation below error.

```

error[E0277]: the trait bound `&&std::collections::HashMap<syntax::ast::Symbol, std::vec::Vec<syntax::ast::Symbol>>: std::iter::Iterator` is not satisfied
   --> src/lib.rs:728:42
    |
728 |             for (ref key, ref values) in &interchangeables {
    |                                          ^^^^^^^^^^^^^^^^^ `&&std::collections::HashMap<syntax::ast::Symbol, std::vec::Vec<syntax::ast::Symbol>>` is not an iterator; maybe try calling `.iter()` or a similar method
    |
    = help: the trait `std::iter::Iterator` is not implemented for `&&std::collections::HashMap<syntax::ast::Symbol, std::vec::Vec<syntax::ast::Symbol>>`
    = note: required by `std::iter::IntoIterator::into_iter`
```

Could you please suggest some ideas how to resolve this error.

Reply:
Ah, my mistake: `interchangeables` already is a reference. In that case, you don't need the `&`.